test/integration/createProvider_test.go

```diff
@@ -0,0 +1,108 @@
+/*
```

Comment:
Nit: Could we name this file `create_provider_test.go` instead? I feel like snake case is the prevalent pattern for Go filenames.